Project/Assets/Scripts/SoilTile.cs

```diff
@@ -8,5 +11,20 @@
     public int WaterLevel;
     public int SunshineLevel;
     public int WindLevel;
+    public int QualityLevel;
     public int Elevation;
+
+    internal void Initialize(int sunshine, int wind)
+    {
+        this.SunshineLevel = sunshine;
+        this.WindLevel = wind;
+        this.WaterLevel = this.Description.InitialWaterLevel;
+        this.QualityLevel = this.Description.InitialQualityLevel;
+    }
+
+    internal void UpdateQuality(int delta)
+    {
+        this.QualityLevel += delta;
+        Mathf.Clamp(this.QualityLevel, 0, 100);
```

Comment:
Il faut stocker la valeur de retour :)
```suggestion
        this.QualityLevel = Mathf.Clamp(this.QualityLevel, 0, 100);
```